pkg/configurations/configurations.go

```diff
@@ -0,0 +1,147 @@
+package configurations
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/nmstate/kubernetes-nmstate/pkg/helper"
+	"github.com/thoas/go-funk"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	nnscontroller "github.com/nmstate/kubernetes-nmstate/pkg/controller/nodenetworkstate"
+	"github.com/spf13/viper"
+)
+
+var log = logf.Log.WithName("configurations")
+
+const defaultConfigPath = "/etc/config/"
+const defaultConfigName = "nmstate-config.yaml"
+const defaultInterfacesFilter = "veth*"
+const defaultNodeNetworkStateRefresh = "5"
+
+var (
+	configPath = defaultConfigPath
+	configName = defaultConfigName
+)
+
+// Take os environment values if given else use defaults
+func init() {
+	configPathTemp, isSet := os.LookupEnv("CONFIG_PATH")
```

Comment:
You put the default for CONFIG_NAME into the manifest, please be consistent and do it with this one too. I think we can drop the defaults from here and just make it required. Require only one parameter - the path to the config, you can get the directory out of it here.